src/consumption.cpp

```diff
@@ -589,6 +592,27 @@ int Character::vitamin_get( const vitamin_id &vit ) const
     return v != vitamin_levels.end() ? v->second : 0;
 }
 
+int Character::get_daily_vitamin( const vitamin_id &vit, bool actual ) const
+{
+    if( get_option<bool>( "NO_VITAMINS" ) && vit->type() == vitamin_type::VITAMIN ) {
+        return 0;
+    }
+
+    const auto &v = daily_vitamins.find( vit );
+    // if we should get the guess or the real value
+    const int &count = actual ? v->second.second : v->second.first;
+    return v != daily_vitamins.end() ? count : 0;
```

Comment:
If v is an iterator to the end, dereferencing it triggers undefined behaviour. And you dereference it in the previous line before that check.